File: flatdata-generator/tests/generators/go_expectations/archives/empty.go
type AArchive struct {
    IsOptional bool
    IsOpen bool
}

func (v *AArchive) Close() {
}

func (v *AArchive) GetSizeInBytes() int {
    var size int
    return size
}

func (v *AArchive) ToString() string {
    buffer := bytes.Buffer{}
    buffer.WriteString(fmt.Sprintf(`{"name": "A", "container_type": "Archive", "size_in_bytes": %d, "resources": [`, v.GetSizeInBytes()))
    buffer.WriteString("]}")
	return buffer.String()
}

func OpenAArchive(resource flatdata.ResourceStorage) (*AArchive, error) {
    v := &AArchive{}
    // Initialize resources
	// Add resources to archive
	return v, nil
}

File: flatdata-generator/tests/generators/py_expectations/archives/empty.py
class n_A(flatdata.archive.Archive):
    _SCHEMA = """namespace n {
archive A
{
}
}

"""
    _NAME = "A"
    _RESOURCES = {
        "A.archive" : flatdata.archive.ResourceSignature(
            container=flatdata.resources.RawData,
            initializer=None,
            schema=_SCHEMA,
            is_optional=False,
            doc="Archive signature"),
    }

    def __init__(self, resource_storage):
        flatdata.archive.Archive.__init__(self, resource_storage)